craft_application/application.py

```diff
@@ -400,6 +411,33 @@ def _register_default_plugins(self) -> None:
         """Register per application plugins when initializing."""
         self.register_plugins(self._get_app_plugins())
 
+    def _pre_run(self, dispatcher: craft_cli.Dispatcher) -> None:
+        """Do any final setup before running the command.
+
+        At the time this is run, the command is loaded in the dispatcher, but
+        the project has not yet been loaded.
+        """
+        # Some commands might have a project_dir parameter. Those commands and
+        # only those commands should get a project directory, but only when
+        # not managed.
+        if self.is_managed():
+            self.project_dir = pathlib.Path("/root/project")
```

Comment:
Since this exists 3 times now, it might be wise to make this a constant, although that doesn't have to be part of this PR.

Reply:
Agreed! https://github.com/canonical/craft-application/issues/195